doc/unet.txt

```diff
@@ -0,0 +1,161 @@
+path, which allows to transfer information. By comparison, the basic FCN architecture only had
+*number of classes* feature maps in its upsampling path.
+
+The U-Net owes its name to its symmetric shape, which is different from other FCN variants.
+
+U-Net architecture is separated in 3 parts:
+
+- 1 : The contracting/downsampling path
+- 2 : Bottleneck
+- 3 : The expanding/upsampling path
+
+.. figure:: images/unet.jpg
+    :align: center
+    :scale: 60%
+
+    **Figure 1** : Illustration of U-Net architecture (from U-Net paper)
+
+
+Contracting/downsampling path
+=============================
+
+The contracting path is composed of 4 blocks. Each block is composed of
+
+* 3x3 Convolution Layer + activation function (with batch normalization)
+* 3x3 Convolution Layer + activation function (with batch normalization)
+* 2x2 Max Pooling
+
+Note that the number of feature maps doubles at each pooling, starting with
+64 feature maps for the first block, 128 for the second, and so on.
+The purpose of this contracting path is to capture the context of the input image
+in order to be able to do segmentation. This coarse contextual information will
+then be transfered to the upsampling path by means of skip connections.
+
+
+Bottleneck
+==========
+
+This part of the network is between the contracting and expanding paths.
+The bottleneck is built from simply 2 convolutional layers (with batch
+normalization), with dropout.
+
+
+Expanding/upsampling path
+=========================
+
+The expanding path is also composed of 4 blocks. Each of these blocks is composed of
+
+* Deconvolution layer with stride 2
+* Concatenation with the corresponding cropped feature map from the contracting path
+* 3x3 Convolution layer + activation function (with batch normalization)
+* 3x3 Convolution layer + activation function (with batch normalization)
+
+
+The purpose of this expanding path is to enable precise localization combined
+with contextual information from the contracting path.
+
+Advantages
+==========
+
+* The U-Net combines the location information from the downsampling path with the contextual information in the upsampling path to finally obtain a general information combining localisation and context, which is necessary to predict a good segmentation map.
+
+* No dense layer, so images of different sizes can be used as input (since the only parameters to learn on convolution layers are the kernel, and the size of the kernel is independent from input image' size).
+
+* The use of massive data augmentation is important in domains like biomedical segmentation, since the number of annotated samples is usually limited.
+
+
+Code
+++++
+
+The U-Net implementation can be found in the following GitHub repo:
+
+* `Unet.py <https://github.com/Lasagne/Recipes/blob/master/modelzoo/Unet.py>`_ : Main script. Defines the model
+
+* `train_unet.py <../code/unet/train_unet.py>`_ : Training loop.
```

Comment:
This one starts correctly as well (needs more memory than I have available now).